### PR TITLE
Refactors all (F) links into using a define

### PR DESCRIPTION
**Pull request by coiax**
:cl:  coiax
bugfix: Clicking the (F) link when an AI talks in binary chat will
follow its camera eye, the same as when (F) is clicked for its radio
chat.
/:cl:

FOLLOW_LINK is a define that creates the clickable (F) link, so it
doesn't have to be typed painstakingly out every time we want to give
that link to an observer.

### CODE REVIEW

File: code/__DEFINES/say.dm [hunk 1]
```diff
@@ -29,3 +29,6 @@
 #define ITALICS 1
 #define REDUCE_RANGE 2
 #define NOPASS 4
+
+// A link given to alice to follow bob
+#define FOLLOW_LINK(alice, bob) "<a href=?src\ref[alice];follow=\ref[bob]>(F)</a>"
```
Comment (ChangelingRain) [inline code/__DEFINES/say.dm]: should note it only works for ghosts

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 2]
```diff
@@ -67,7 +67,8 @@
 		if(isovermind(M) || istype(M, /mob/living/simple_animal/hostile/blob))
 			M << rendered
 		if(isobserver(M))
-			M << "<a href='?src=\ref[M];follow=\ref[src]'>(F)</a> [rendered]"
+			var/link = FOLLOW_LINK(M, src)
+			M << "[link] [rendered]"
```
Comment (tigercat2000) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: to_chat(M, "[link] [rendered]")

File: code/game/gamemodes/blob/overmind.dm [hunk 3]
```diff
@@ -135,7 +135,8 @@
 		if(isovermind(M) || istype(M, /mob/living/simple_animal/hostile/blob))
 			M << rendered
 		if(isobserver(M))
-			M << "<a href='?src=\ref[M];follow=\ref[src]'>(F)</a> [rendered]"
+			var/link = FOLLOW_LINK(M, src)
+			M << "[link] [rendered]"
```
Comment (tigercat2000) [inline code/game/gamemodes/blob/overmind.dm]: to_chat(M, "[link] [rendered]")

**Comment by Bawhoppen** [comment]
why would I want to follow the camera eye instead of the core itself?

**Comment by coiax** [comment]
Because the AI doesn't move. Radio code already follows the eye, because its what the AI is looking at/interacting with.

**Comment by Bawhoppen** [comment]
yes but the core is what actually matters, its how the AI is killed, why would you want to look at where the AI is looking over it's actual body?

**Comment by coiax** [comment]
If you care about whether the AI is dying or not, you're following the syndiborg attacking it, or the operation to bust into its core. When you click (F) when it talks about HARM IN CARGO, you don't want to see it, you want to see where it is looking. This is just bringing it inline with what the radio does.

**Comment by Bawhoppen** [comment]
I think most people would agree with me, lets make a player poll

**Comment by coiax** [comment]
I don't want an important structural PR delayed because of minor bikeshedding.

**Comment by Jordie0608** [comment]
Following the eye is fine

**Comment by lzimann** [comment]
i don't think most players agree with you though

File: code/modules/mob/living/silicon/say.dm [hunk 4]
```diff
@@ -18,7 +18,14 @@
 			else
 				M << rendered
 		if(isobserver(M))
-			M << "<a href='?src=\ref[M];follow=\ref[src]'>(F)</a> [rendered]"
+			var/following = src
+			// If the AI talks on binary chat, we still want to follow
+			// it's camera eye, like if it talked on the radio
+			if(istype(src, /mob/living/silicon/ai))
+				var/mob/living/silicon/ai/ai = M
```
Comment (phil235) [inline code/modules/mob/living/silicon/say.dm]: = src , not  = M  , this causes a runtime. @coiax

**Pull request merged**